<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Sekai Framework: module::Exception Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.6.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="classes.html"><span>Class&nbsp;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath"><a class="el" href="namespacemodule.html">module</a>::<a class="el" href="classmodule_1_1_exception.html">Exception</a>
  </div>
</div>
<div class="contents">
<h1>module::Exception Class Reference<br/>
<small>
[<a class="el" href="group___common.html">Common</a>]</small>
</h1><!-- doxytag: class="module::Exception" -->
<p>Base exception class.  
<a href="#_details">More...</a></p>

<p><code>#include &lt;<a class="el" href="exception_8h_source.html">exception.h</a>&gt;</code></p>
<div class="dynheader">
Collaboration diagram for module::Exception:</div>
<div class="dynsection">
<div class="center"><img src="classmodule_1_1_exception__coll__graph.png" border="0" usemap="#module_1_1_exception_coll__map" alt="Collaboration graph"/></div>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>

<p><a href="classmodule_1_1_exception-members.html">List of all members.</a></p>
<table border="0" cellpadding="0" cellspacing="0">
<tr><td colspan="2"><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#aa8f79144026e34f3a96ffd00cea6fa6d">Exception</a> (const char *msg, int code=0)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a26009f58fafbf316f69cff43334010dc">Exception</a> (const char *msg, const char *arg, int code=0)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#ae1de1e6ec8fd2fa1b84db7c84729c217">Exception</a> (const char *msg, const <a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp;nested, int code=0)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">clones the nested exception  <a href="#ae1de1e6ec8fd2fa1b84db7c84729c217"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#afa11be5194e2b536cfcff0527fc267b8">Exception</a> (const <a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp;other)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a4f30c5c64cb2dedf6e988ee3bb0e9707">~Exception</a> ()  throw ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a9491af8e0104f7f65fbfebfb76145b1a">operator=</a> (const <a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp;exc)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">virtual const char *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a60d12a7328b76ab996be3ece1163d85f">className</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Returns the name of the exception class.  <a href="#a60d12a7328b76ab996be3ece1163d85f"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">const char *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a84e9f78cd96533192a0bd5b90090abf7">message</a> () const </td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Returns a static string describing the exception.  <a href="#a84e9f78cd96533192a0bd5b90090abf7"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">virtual const char *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a845060e6c078848727c9ede37f902680">what</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Same as <a class="el" href="classmodule_1_1_exception.html#a84e9f78cd96533192a0bd5b90090abf7" title="Returns a static string describing the exception.">message()</a>, for compatibility with std::exception.  <a href="#a845060e6c078848727c9ede37f902680"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">const <a class="el" href="classmodule_1_1_exception.html">Exception</a> *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#aafd9cc797c6e73972816a9538f36d44a">nested</a> () const </td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Returns a pointer to the nested exception.  <a href="#aafd9cc797c6e73972816a9538f36d44a"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">int&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a9b3d38706fe604a8ee7bd9fd2b9f1ae1">code</a> () const </td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Returns the exception code if defined.  <a href="#a9b3d38706fe604a8ee7bd9fd2b9f1ae1"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">virtual <a class="el" href="classmodule_1_1_exception.html">Exception</a> *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a7d9f45e40edf626d8a1deb60411a338c">clone</a> () const </td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Creates an exact copy of the exception.  <a href="#a7d9f45e40edf626d8a1deb60411a338c"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">virtual void&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a1ff6f5707a03e84eed7c250cb5f7ac96">release</a> ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Releases exception, use only on cloned ones.  <a href="#a1ff6f5707a03e84eed7c250cb5f7ac96"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">virtual void&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmodule_1_1_exception.html#a5ed83fa27c585c0aeeb83cde3e76d370">rethrow</a> () const </td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">(Re)Throws the exception.  <a href="#a5ed83fa27c585c0aeeb83cde3e76d370"></a><br/></td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>
<p>Base exception class. </p>

<p>Definition at line <a class="el" href="exception_8h_source.html#l00020">20</a> of file <a class="el" href="exception_8h_source.html">exception.h</a>.</p>
<hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="aa8f79144026e34f3a96ffd00cea6fa6d"></a><!-- doxytag: member="module::Exception::Exception" ref="aa8f79144026e34f3a96ffd00cea6fa6d" args="(const char *msg, int code=0)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">module::Exception::Exception </td>
          <td>(</td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>msg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&nbsp;</td>
          <td class="paramname"> <em>code</em> = <code>0</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00017">17</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a26009f58fafbf316f69cff43334010dc"></a><!-- doxytag: member="module::Exception::Exception" ref="a26009f58fafbf316f69cff43334010dc" args="(const char *msg, const char *arg, int code=0)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">module::Exception::Exception </td>
          <td>(</td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>msg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>arg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&nbsp;</td>
          <td class="paramname"> <em>code</em> = <code>0</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00027">27</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="ae1de1e6ec8fd2fa1b84db7c84729c217"></a><!-- doxytag: member="module::Exception::Exception" ref="ae1de1e6ec8fd2fa1b84db7c84729c217" args="(const char *msg, const Exception &amp;nested, int code=0)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">module::Exception::Exception </td>
          <td>(</td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>msg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>nested</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&nbsp;</td>
          <td class="paramname"> <em>code</em> = <code>0</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>clones the nested exception </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00044">44</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="afa11be5194e2b536cfcff0527fc267b8"></a><!-- doxytag: member="module::Exception::Exception" ref="afa11be5194e2b536cfcff0527fc267b8" args="(const Exception &amp;other)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">module::Exception::Exception </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>other</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00054">54</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a4f30c5c64cb2dedf6e988ee3bb0e9707"></a><!-- doxytag: member="module::Exception::~Exception" ref="a4f30c5c64cb2dedf6e988ee3bb0e9707" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">module::Exception::~Exception </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td>  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00064">64</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="a60d12a7328b76ab996be3ece1163d85f"></a><!-- doxytag: member="module::Exception::className" ref="a60d12a7328b76ab996be3ece1163d85f" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const char * module::Exception::className </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()<code> [virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Returns the name of the exception class. </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00096">96</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a7d9f45e40edf626d8a1deb60411a338c"></a><!-- doxytag: member="module::Exception::clone" ref="a7d9f45e40edf626d8a1deb60411a338c" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classmodule_1_1_exception.html">Exception</a> * module::Exception::clone </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Creates an exact copy of the exception. </p>
<p>The copy can later be thrown again by invoking <a class="el" href="classmodule_1_1_exception.html#a5ed83fa27c585c0aeeb83cde3e76d370" title="(Re)Throws the exception.">rethrow()</a> on it </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00110">110</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a9b3d38706fe604a8ee7bd9fd2b9f1ae1"></a><!-- doxytag: member="module::Exception::code" ref="a9b3d38706fe604a8ee7bd9fd2b9f1ae1" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int module::Exception::code </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Returns the exception code if defined. </p>

<p>Definition at line <a class="el" href="exception_8h_source.html#l00088">88</a> of file <a class="el" href="exception_8h_source.html">exception.h</a>.</p>

</div>
</div>
<a class="anchor" id="a84e9f78cd96533192a0bd5b90090abf7"></a><!-- doxytag: member="module::Exception::message" ref="a84e9f78cd96533192a0bd5b90090abf7" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const char * module::Exception::message </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Returns a static string describing the exception. </p>

<p>Definition at line <a class="el" href="exception_8h_source.html#l00081">81</a> of file <a class="el" href="exception_8h_source.html">exception.h</a>.</p>

</div>
</div>
<a class="anchor" id="aafd9cc797c6e73972816a9538f36d44a"></a><!-- doxytag: member="module::Exception::nested" ref="aafd9cc797c6e73972816a9538f36d44a" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const <a class="el" href="classmodule_1_1_exception.html">Exception</a> * module::Exception::nested </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Returns a pointer to the nested exception. </p>

<p>Definition at line <a class="el" href="exception_8h_source.html#l00074">74</a> of file <a class="el" href="exception_8h_source.html">exception.h</a>.</p>

</div>
</div>
<a class="anchor" id="a9491af8e0104f7f65fbfebfb76145b1a"></a><!-- doxytag: member="module::Exception::operator=" ref="a9491af8e0104f7f65fbfebfb76145b1a" args="(const Exception &amp;exc)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp; module::Exception::operator= </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classmodule_1_1_exception.html">Exception</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>exc</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00073">73</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a1ff6f5707a03e84eed7c250cb5f7ac96"></a><!-- doxytag: member="module::Exception::release" ref="a1ff6f5707a03e84eed7c250cb5f7ac96" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void module::Exception::release </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td><code> [virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Releases exception, use only on cloned ones. </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00117">117</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a5ed83fa27c585c0aeeb83cde3e76d370"></a><!-- doxytag: member="module::Exception::rethrow" ref="a5ed83fa27c585c0aeeb83cde3e76d370" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void module::Exception::rethrow </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>(Re)Throws the exception. </p>
<p>This is useful for temporarily storing a copy of an exception, then throwing it again </p>
<dl class="see"><dt><b>See also:</b></dt><dd><a class="el" href="classmodule_1_1_exception.html#a7d9f45e40edf626d8a1deb60411a338c" title="Creates an exact copy of the exception.">clone</a> </dd></dl>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00124">124</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a845060e6c078848727c9ede37f902680"></a><!-- doxytag: member="module::Exception::what" ref="a845060e6c078848727c9ede37f902680" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const char * module::Exception::what </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()<code> [virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Same as <a class="el" href="classmodule_1_1_exception.html#a84e9f78cd96533192a0bd5b90090abf7" title="Returns a static string describing the exception.">message()</a>, for compatibility with std::exception. </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00103">103</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li>D:/Programming/Projects/depot/SekaiFramework/src/module/<a class="el" href="exception_8h_source.html">exception.h</a></li>
<li>D:/Programming/Projects/depot/SekaiFramework/src/module/implementation/<a class="el" href="exception_8cpp_source.html">exception.cpp</a></li>
</ul>
</div>
<hr size="1"/><address style="text-align: right;"><small>Generated on Wed Apr 28 21:26:52 2010 for Sekai Framework by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.6.1 </small></address>
</body>
</html>
